### PR TITLE
Added 'filter' alias for find_all and select.

### PR DESCRIPTION
**Pull request by adp90**
As per Feature [#13784](https://bugs.ruby-lang.org/issues/13784).

### CODE REVIEW

**Comment by davidarnold** [comment]
Not sure, but I think there should be an update to the specs too.

```
require File.expand_path('../../../spec_helper', __FILE__)
require File.expand_path('../fixtures/classes', __FILE__)
require File.expand_path('../shared/find_all', __FILE__)

describe "Enumerable#filter" do
  it_behaves_like(:enumerable_find_all , :filter)
end
```

**Comment by adp90** [comment]
Added the spec, thanks.
I also added filter to enumerator.c.

Functionality after make && make install:
[1,2,3,4,5].filter { |num| num.even? } #=> [2, 4]
lazy_filter = (0..Float::INFINITY).lazy.filter(&:odd?) #=> #<Enumerator::Lazy: 0..Infinity>:filter>
lazy_filter.next #=> 1
lazy_filter.next #=> 3
lazy_filter.next #=> 5

File: spec/rubyspec/core/enumerable/filter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+require File.expand_path('../fixtures/classes', __FILE__)
+require File.expand_path('../shared/find_all', __FILE__)
```
Comment (bogdanvlviv) [inline spec/rubyspec/core/enumerable/filter_spec.rb]: could you please use `require_relative`?

Comment (eregon) [inline spec/rubyspec/core/enumerable/filter_spec.rb]: No require_relative in specs.
If we update it, we should update it for all specs (in https://github.com/ruby/spec/) to remain consistent.

New specs can be directly here though, that's perfectly fine.

**Pull request closed without merging**